### PR TITLE
Waitlist registration entry 

### PR DESCRIPTION
**Pull request by dhmi19**
## Notion ticket link

[Waitlist Registration: Invite Waitlisted camper to register](https://www.notion.so/uwblueprintexecs/Task-Board-F22-9e03bf9f7c4343f09a8a5c16dc48a12a?p=74da1c05846d4a719c861bca84010cc1&pm=c)

## Implementation description
* changed the model for waitlisted camper so that the contact is split into firstname and last name. This is because the registrant flow needs the primary and secondary contact to have 2 fields for firstname and lastname. 

## Steps to test
1. Go to `http://localhost:3000/register/camp/6397f94e2ff11dcda0ba4960?waitlistedSessionId=6397fa49ef6ca9aaa10cff62&waitlistedCamperId=63ab387d4a2c7a5783a22c94` 
2. You can change the session Id, and mess around with the waitlisted camper on the db like changing expiry date to see that we redirect with error toast 
3. Try testing with other camps too 

## What should reviewers focus on?
* 

## Checklist
- [ ] My PR name is descriptive and in imperative tense
- [ ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/pages/RegistrantExperience/RegistrationSteps/RegistrationFooter.tsx [hunk 1]
```diff
@@ -39,16 +41,19 @@ const RegistrationFooter = ({
...
-        height="48px"
-        variant="secondary"
-        onClick={() => handleStepNavigation(-1)}
-        mb={{ sm: 4, md: 0 }}
-        mr={{ sm: 0, md: 4 }}
-      >
-        Back
-      </Button>
+      {(!isWaitlistRegistration ||
+        currentStep !== RegistrantExperienceSteps.PersonalInfoPage) && (
```
Comment (e-wai) [inline frontend/src/components/pages/RegistrantExperience/RegistrationSteps/RegistrationFooter.tsx:45]: This logic is somewhat difficult to parse, can we write this as `!(isWaitListRegistration && currentStep === Reg...PersonalInfoPage)` maybe?

File: frontend/src/components/pages/RegistrantExperience/index.tsx [hunk 2]
```diff
@@ -1,21 +1,29 @@
...
   camp: boolean;
 };
 
 const RegistrantExperiencePage = (): React.ReactElement => {
+  const toast = useToast();
+  const history = useHistory();
+
+  const location = useLocation();
+  const searchParams = new URLSearchParams(location.search);
   const { id: campId } = useParams<{ id: string }>();
```
Comment (e-wai) [inline frontend/src/components/pages/RegistrantExperience/index.tsx:24]: nit: switch order of these two lines

File: frontend/src/components/pages/RegistrantExperience/index.tsx [hunk 3]
```diff
@@ -35,10 +43,53 @@ const RegistrantExperiencePage = (): React.ReactElement => {
...
+              throw new Error("The invitation link has expired");
+            }
+            setSessionSelectionIsComplete(true);
+            setWaitlistedCamper(waitlistCamper);
+            setSelectedSessions(new Set(waitlistedSessionId));
+          }
+        }
+      })
+      .catch((error) => {
+        history.push(`/error`);
```
Comment (e-wai) [inline frontend/src/components/pages/RegistrantExperience/index.tsx:85]: we need a prettier error page LOL

**Comment by e-wai** [comment]
Should "relation to camper" be populated automatically? Mine isn't rn

**Comment by e-wai** [comment]
Any thoughts?

Screen Shot 2023-01-01 at 10 47 41 PM

**Comment by dhmi19** [comment]
> Should "relation to camper" be populated automatically? Mine isn't rn

no, it's not on the model of a waitlisted camper and we don't collect that info from the parent when they are waitlisting

**Comment by e-wai** [comment]
Using - http://localhost:3000/register/camp/6397f94e2ff11dcda0ba4960?waitlistedCamperId=63ab387d4a2c7a5783a22c94&waitlistedSessionId=6397fa49ef6ca9aaa10cff62

nit: maybe button right aligned? 

Screen Shot 2023-01-03 at 11 03 48 PM

are cart items supposed to be working yet?

Screen Shot 2023-01-03 at 11 04 28 PM

File: frontend/src/components/pages/RegistrantExperience/RegistrationSteps/ReviewRegistration/index.tsx [hunk 4]
```diff
@@ -1,7 +1,6 @@
 import React, { useEffect } from "react";
 import { Box } from "@chakra-ui/react";
 import PaymentSummary from "./PaymentSummary";
-import ReviewInformation from "./ReviewInformation";
```
Comment (e-wai) [inline frontend/src/components/pages/RegistrantExperience/RegistrationSteps/ReviewRegistration/index.tsx:4]: How did you make it compile even though the component isn't used?

**Review by e-wai — APPROVED** [review]
^ just that one question about cart times and how you want to handle checkout, otherwise looks good

File: frontend/src/components/pages/RegistrantExperience/RegistrationSteps/ReviewRegistration/index.tsx (hunk 4 above)
Comment (dhmi19) [reply frontend/src/components/pages/RegistrantExperience/RegistrationSteps/ReviewRegistration/index.tsx:4]: review registration is currently commented out. We add this in tejas's PR

**Comment by dhmi19** [comment]
Pushed a commit to fix the cart item not showing up. 2 changes:

1. I was setting the selected session set properly so the session passed in was never "selected"
2. The mapToCartItems function assumed that there will always be EDLP selected so I added some checks to make sure EDLP was actually selected before populating the cart item

**Pull request merged**